### DISCUSSION TITLE
Ability to reduce the max download speed

### DISCUSSION THREAD

**Discussion by CesarCruzUtec** [BlafKing/sd-civitai-browser-plus#168] [Ideas]
It would be really helpful to be able to reduce the download speed, since I would like to do other things while downloading checkpoints or many loras. We are a family of 4 so when I download some checkpoints the other members have lag problems including myself

**Comment by BlafKing** [comment]
Thanks for posting!
This is possible if you use Aria2 to download files (which is the default download method)

You can go to the settings tab of SD-WebUI and then find the `CivitAI Browser+` tab.
(If you're using SD-Webui v1.7.0+ you need to select the `Downloads` tab under the `CivitAI Browser+` header)

Here you can find a setting called `Custom Aria2 command line flags (Requires UI reload)`
You can use the following Aria2 flag to limit your download speed:

`--max-overall-download-limit=<SPEED>`
This sets the overall max download speed in bytes/sec. 0 means unrestricted. You can append K or M (1K = 1024, 1M = 1024K). To limit the download speed per download

[Here you can find all Aria2 flags](https://aria2.github.io/manual/en/html/aria2c.html)

Here's an example where I limit the download speed to 20mb/s:
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/9644716/7d3dd9d6-87d8-433d-b10c-4cab60f5c61a)

After you filled in this setting, press the `Apply settings` button and then `Reload UI` for it to take effect! :)

**Reply by CesarCruzUtec** [reply]
Oh, thank you, I don´t know about aria and its commands 😅, maybe a setting for it would be a good qol for some people.
